writer/writer/postgresql_backend/sql_occ_locker_backend_service.py

```diff
@@ -58,8 +58,12 @@ def assert_fqfield_positions(self, fqfields: Dict[str, int]) -> None:
             )
             event_filter_parts.append("(fqid=%s and position>%s)")
 
+            # _ is the postgres wild card for a single character so we have to escape
+            # all underscores
             collectionfield = collectionfield.replace("_", r"\_")
-            collectionfield = collectionfield.replace("$", "_%")
+            # % matches zero or more chars: this is correct since the template field
+            # itself may also be locked
+            collectionfield = collectionfield.replace("$", "$%")
```

Comment:
I wonder why this works. If I have `group_$2_ids`, there should be no `%`, only for `group_$_ids`. Maybe test this:

`a_$1_f` modified, `a_$11_f` locked -> ok

Reply:
Maybe I need an explanation here ;)

Reply:
Ah, you thought the wrong way: the failing test is `a_$11_f` modified, `a_$1_f` locked -> ok. (since the locked fqfields are checked in the db, not the modified ones). Trying to fix that. But good catch

Reply:
Right! I meant this ;)

Reply:
I was curios since I knew when changing the syntax that we must somewhere differentiate `$_` from`$[a-z...]`. And I missed that point in the implementation.

Reply:
Should be fixed now